src/ploigos_step_runner/step_implementers/deploy/argocd_deploy.py

```diff
@@ -1,16 +1,224 @@
-"""`StepImplementer` for the `package` step using npm
+"""`StepImplementer` for the `deploy` step using ArgoCD.
 
-"""
+Step Configuration
+------------------
+Step configuration expected as input to this step.
+Could come from:
+
+  * static configuration
+  * runtime configuration
+  * previous step results
+
+Configuration Key         | Required? | Default  | Description
+--------------------------|-----------|----------|---------------------------
+`argocd-username`         | Yes       |          | Username for accessing the ArgoCD API
+`argocd-password`         | Yes       |          | Password for accessing the ArgoCD API
+`argocd-api`              | Yes       |          | The ArgoCD API endpoint
```

Comment:
@itewk, the configuration keys argocd-username, argocd-password, and argocd-api will be need by any feature/subclass that needs to communicate with the ArgoCD server. You may want to consider defining it in the generic class.

Method: ArgoCDGeneric::_argocd_sign_in

Reply:
@gfulton-redhat they deff should be, but hence my note in the description of this one that some of this stuff shoudl be moved to generic, but waiting for first other sub class to do that. for now need this minimum refactor for some other stuf fin works.